src/query/storage/m3/cluster_resolver_test.go

```diff
@@ -403,3 +403,51 @@ func TestLongUnaggregatedRetention(t *testing.T) {
 	assert.Equal(t, expected, actualNames)
 	assert.Equal(t, namespaceCoversPartialQueryRange, fanoutType)
 }
+
+func TestExampleCase(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	session := client.NewMockSession(ctrl)
+	ns, err := NewClusters(
+		UnaggregatedClusterNamespaceDefinition{
+			NamespaceID: ident.StringID("metrics_10s_24h"),
+			Retention:   24 * time.Hour,
+			Session:     session,
+		}, AggregatedClusterNamespaceDefinition{
+			NamespaceID: ident.StringID("metrics_180s_360h"),
+			Retention:   360 * time.Hour,
+			Resolution:  120 * time.Second,
+			Downsample:  &ClusterNamespaceDownsampleOptions{All: false},
+			Session:     session,
+		}, AggregatedClusterNamespaceDefinition{
+			NamespaceID: ident.StringID("metrics_600s_17520h"),
+			Retention:   17520 * time.Hour,
+			Resolution:  600 * time.Second,
+			Downsample:  &ClusterNamespaceDownsampleOptions{All: false},
+			Session:     session,
+		},
+	)
+	require.NoError(t, err)
+
+	now := time.Now()
+	end := now
+
+	for i := 27; i < 17520; i++ {
```

Comment:
Magic numbers? What do they mean?

Reply:
ah, it's to ensure we hit a retention out of the unaggregated namespace but within `metrics_600s_17520h `